src/GraphQL.Tests/Subscription/ShouldlyExtensions.cs

```diff
@@ -0,0 +1,63 @@
+#nullable enable
+
+using GraphQL.Subscription;
+using GraphQL.SystemTextJson;
+
+namespace GraphQL.Tests.Subscription;
+
+public static class ShouldlyExtensions
+{
+    private static readonly IGraphQLTextSerializer _serializer = new GraphQLSerializer();
+
+    public static void ShouldBeSimilarTo(this object? actual, object? expected, string? customMessage = null)
+    {
+        if (expected is string str)
+            expected = _serializer.Deserialize<Inputs>(str);
+        var expectedJson = _serializer.Serialize(expected);
+        var actualJson = _serializer.Serialize(actual);
+        actualJson.ShouldBe(expectedJson, customMessage);
+    }
+
+    public static ExecutionResult ShouldHaveResult(this SampleObserver? observer)
+    {
+        observer.ShouldNotBeNull();
+        observer.Events.ShouldNotBeNull();
+        observer.Events.TryDequeue(out var result).ShouldBeTrue("Observable sequence should have another result but did not");
+        return result;
+    }
+
+    public static void ShouldHaveNoMoreResults(this SampleObserver? observer)
+    {
+        observer.ShouldNotBeNull();
+        observer.Events.ShouldNotBeNull();
+        observer.Events.TryDequeue(out var result).ShouldBeFalse("Observable sequence should not have additional results but did");
+    }
+
+    public static ExecutionResult ShouldNotBeSuccessful(this ExecutionResult result)
+    {
+        if (result is SubscriptionExecutionResult subscriptionExecutionResult)
+        {
+            subscriptionExecutionResult.Streams.ShouldBeNull();
+        }
+        result.Data.ShouldBeNull();
```

Comment:
```suggestion
            result.Data.ShouldBeNull();
        }
```
?

Reply:
It does not matter because of further removal of `SubscriptionExecutionResult`.

Reply:
Then I'll catch it in the subsequent PR.